src/Fixer/PhpUnit/PhpUnitSetUpTearDownVisibilityFixer.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\PhpUnit;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Indicator\PhpUnitTestCaseIndicator;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Gert de Pagter
+ */
+final class PhpUnitSetUpTearDownVisibilityFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Changes the visibility of the setUp and tearDown functions of phpunit to protected, to match its parent.',
+            [
+                new CodeSample(
+                    '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    private $hello;
+    public function setUp()
+    {
+        $this->hello = "hello";
+    }
+
+    public function tearDown()
+    {
+        $this->hello = null;
+    }
+}
+'
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_FUNCTION]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach (array_reverse($this->findPhpUnitClasses($tokens)) as $indexes) {
```

Comment:
Maybe it is more efficient to search backwards and store the result, than the `array_reverse` is not needed.
Even more optimized code is possible by using `yield` in the `findPhpUnitClasses`, but this is not the most important optim here ;)

Reply:
I'm not quite sure how to do that, so if possible, i'd like some help with that.

Reply:
the easiest way would be to iterate backwards in `findPhpUnitClasses`,

```php
for ($index = $tokens->count() - 1; $index >= 0; --$index) {
```

there should be bunch of fixers that do this already which might help as an example. Maybe I can help you out later with this.